src/aap_eda/services/ruleset/activation_podman.py

```diff
@@ -93,75 +97,22 @@ def run_worker_mode(
         activation_instance: models.ActivationInstance,
         heartbeat: int,
         ports: dict,
+        reconnect: bool,
     ) -> None:
-        container = None
         try:
-            """Run ansible-rulebook in worker mode."""
-            args = [
-                "ansible-rulebook",
-                "--worker",
-                "--websocket-ssl-verify",
-                ws_ssl_verify,
-                "--websocket-address",
-                ws_url,
-                "--id",
-                str(activation_instance.id),
-                "--heartbeat",
-                str(heartbeat),
-            ]
-            if (
-                settings.ANSIBLE_RULEBOOK_LOG_LEVEL
-                and settings.ANSIBLE_RULEBOOK_LOG_LEVEL in VALID_LOG_LEVELS
-            ):
-                args.append(settings.ANSIBLE_RULEBOOK_LOG_LEVEL)
-
-            self.pod_args[
-                "name"
-            ] = f"eda-{activation_instance.id}-{uuid.uuid4()}"
-            if ports:
-                self.pod_args["ports"] = ports
-            self._load_extra_args()
-            self.activation_db_logger.write("Starting Container", True)
-            self.activation_db_logger.write(f"Container args {args}", True)
-            container = self.client.containers.run(
-                image=self.decision_environment.image_url,
-                command=args,
-                stdout=True,
-                stderr=True,
-                remove=True,
-                detach=True,
-                **self.pod_args,
-            )
-
-            logger.info(
-                f"Created container: "
-                f"name: {container.name}, "
-                f"id: {container.id}, "
-                f"ports: {container.ports}, "
-                f"status: {container.status}, "
-                f"command: {args}"
-            )
-
-            now = timezone.now()
-            activation_instance.status = ActivationStatus.RUNNING
-            activation_instance.updated_at = now
-            activation_instance.activation_pod_id = container.id
-            activation_instance.save(
-                update_fields=["status", "updated_at", "activation_pod_id"]
-            )
-
-            activation = activation_instance.activation
-            activation.status = ActivationStatus.RUNNING
-            activation.is_valid = True
-            activation.status_updated_at = now
-            activation.save(
-                update_fields=[
-                    "status",
-                    "status_updated_at",
-                    "is_valid",
-                    "modified_at",
-                ]
-            )
+            container = None
```

Comment:
This initialization must happen outside the try-except block because the variable is used in the `finally` section and if some exception is caught the variable is unbound and we will have an unexpected NameError exception.